AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m

```diff
@@ -502,23 +511,190 @@ - (void)testPerformCosmosDbAsyncOperationWithHttpClient {
         completionHandler(actualData, nil, nil);
       });
   NSDictionary *dic = @{@"abv" : @1, @"foo" : @"bar"};
+  MSMockDocument *mockDoc = [[MSMockDocument alloc] initFromDictionary:dic];
   __block NSData *data = [NSJSONSerialization dataWithJSONObject:dic options:0 error:nil];
 
   // When
   [MSCosmosDb performCosmosDbAsyncOperationWithHttpClient:httpClient
                                               tokenResult:tokenResult
                                                documentId:kMSDocumentIdTest
                                                httpMethod:kMSHttpMethodGet
-                                                     body:data
+                                                 document:mockDoc
                                         additionalHeaders:nil
+                                        additionalUrlPath:kMSDocumentIdTest
                                         completionHandler:handler];
 
+  NSError *error;
+  NSDictionary *dictionary = [NSJSONSerialization JSONObjectWithData:(NSData *)actualData options:0 error:&error];
+  NSDictionary *document = dictionary[@"document"];
+  NSData *actualDocumentAsData = [NSJSONSerialization dataWithJSONObject:document options:0 error:nil];
+
   // Then
   XCTAssertTrue(completionHandlerCalled);
-  XCTAssertEqualObjects(data, actualData);
+  XCTAssertEqualObjects(data, actualDocumentAsData);
   XCTAssertEqualObjects(expectedURLString, [actualURL absoluteString]);
 }
 
+- (void)testPerformCosmosDbAsyncOperationWithNilDocument {
```

Comment:
This unit test seems a big than what it needs to be. Can you explain why you call `performCosmosDbAsyncOperationWithHttpClient:tokenResult:documentId:httpMethod:document:additionalHeaders:additionalUrlPath:completionHandler:` method twice here?

Reply:
resolved